src/test/test_matcher.cpp

```diff
@@ -24,8 +24,8 @@
 #include <format>
 #include <regex>
 
-TestMatcher::TestMatcher(const FilterConfig& config)
-: config_(config)
+TestMatcher::TestMatcher(FilterConfig config)
```

Comment:
If we like this pattern, we should make FilterConfig move-only (we have the `#define DISALLOW_COPY(TypeName)                  
`  macro for this, etc.